src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -182,6 +191,82 @@ private void PopulateAssemblies()
             }
         }
 
+        /// <summary>
+        /// Update FileDefinitions and FileDependencies to pretend that certain Reference items
+        /// explicitly added by a .targets file in the NETStandard.Library are actually normal
+        /// package items. This allows them to show up properly under the "SDK" node in Solution
+        /// Explorer, rather than just under the "Assemblies" node.
+        ///
+        /// This is not meant to be a general mechanism for injecting files that can't be handled
+        /// through the normal NuGet package mechanisms. For that, package authors should be
+        /// creating FileDefinition and FileDependency items instead of Reference items.
```

Comment:
I still don't want to encourage manipulating FileDefinition and FileDependency here as I'm still convinced that we want to remove FileDefinition and FileDependency from all build logic and only promise that the output items of this task remain. There is a lot of perf to be gained if we don't produce all these items on every build.

I'd rather bless this path as the general mechanism. Basically, if you're not adding references from the assets file, then don't fake up the assets file data structure and pretend like you are. Instead, just annotate who added the `Reference`s the old-school way, and we'll take of putting together a tree view with both kinds of references.